packages/wrangler/src/api/startDevWorker/utils.ts

```diff
@@ -56,14 +56,11 @@ export function urlFromParts(
 
 type UnwrapHook<H> = H extends Hook<infer T> ? T : never;
 export function unwrapHook<
-	H extends AsyncHook<T, Args> | undefined,
+	H extends AsyncHook<T, Args>,
 	T extends HookValues = UnwrapHook<H>,
 	Args extends unknown[] = [],
->(
-	hook: H,
-	...args: Args
-): H extends undefined ? UnwrapHook<H> | undefined : UnwrapHook<H> {
-	return typeof hook === "function" ? hook(...args) : hook;
+>(hook: H, ...args: Args) {
+	return (typeof hook === "function" ? hook(...args) : hook) as UnwrapHook<H>;
```

Comment:
Somewhere along the way the TypeScript version got updated and that meant the types in this function started failing. Have updated so they pass but needs confirming that these are correct.